Add decorator to make links and images absolute

chrismytton opened the pull request:
This adds a decorator that when used will make all `<a>` and/or `<img />` elements have absolute urls in their `href` and `src` attributes respectively.

Fixes https://github.com/everypolitician/scraped/issues/38

# Notes to reviewer

I couldn't decide whether it would be better to have three separate decorators (one for images, one for links and one that is a composite of both) or just one that does both images and links. I decided to go for the simplest option and just create one decorator for now, but I'm open to changing it if there's an obvious use case for that.

chrismytton committed:
Add README docs for absolute links and images

chrismytton committed:
Add Scraped::Response::Decorator::AbsoluteUrls class

This decorator is responsible for making all the links and images on a
page absolute.

chrismytton committed:
Handle empty links and images correctly

When link href attributes or image src attributes are empty we should
just leave them alone rather than filling them in with the base url.

tmtmtmtm reviewed (changes requested):
👍  for implementation and tests, but the README is a little bit odd now. You describe this as a decorator before Decorators have been introduced as a concept, and then the later explanation of Decorators gives as an example "you might want to make all the links on the page absolute rather than relative" in a way that's now slightly odd.

chrismytton committed:
Re-order decorator docs in the README

This was trying to introduce the `AbsoluteUrls` decorator before the
concept of decorators had been introduced. I've now moved it to be part
of the "Extending" section under a "Built in decorators" subsection.

chrismytton committed:
Update README "Extending" docs

This links to the new built in decorators section, which serves as a
nice example of the kind of things decorators can be used for.

tmtmtmtm reviewed (approved):
Woot!

I note that we don't have a changelog yet, though perhaps we should…
